lib/puppet/parser/functions/fqdn_rand.rb

```diff
@@ -26,10 +26,12 @@
     # Restoring previous fqdn_rand behavior of calculating its seed value using MD5
     # when running on a non-FIPS enabled platform and only using SHA256 on FIPS enabled
     # platforms.
+    # First convert to lowercase, as DNS hostnames are case-insensitive.
+    longstring = [self['::fqdn'].to_s.downcase,max,args].join(':')
```

Comment:
I think this is the correct thing to do but due to Hyrum’s Law I think this could cause some unexpected behavior if hosts change “slots” similar to when changed from md5 to sha256 on non fips hosts. I’d recommend adding an optional argument to the method to opt o to the new behavior